numpy/fft/_pocketfft.py

```diff
@@ -730,9 +734,11 @@ def fftn(a, s=None, axes=None, norm=None):
         (``s[0]`` refers to axis 0, ``s[1]`` to axis 1, etc.).
         This corresponds to ``n`` for ``fft(x, n)``.
         Along any axis, if the given shape is smaller than that of the input,
-        the input is cropped.  If it is larger, the input is padded with zeros.
-        if `s` is not given, the shape of the input along the axes specified
+        the input is cropped. If it is larger, the input is padded with zeros.
+        If it is ``-1``, the whole input is used (no padding/trimming).
+        If `s` is not given, the shape of the input along the axes specified
         by `axes` is used.
+        If `s` is not ``None``, `axes` must not be ``None`` either.
     axes : sequence of ints, optional
         Axes over which to compute the FFT.  If not given, the last ``len(s)``
```

Comment:
Just a fly-by comment - if we go this route, this part of the docstring needs to be adjusted too.

Reply:
Ah yes, agreed - the spec says

> If `s` is specified, the corresponding `axes` to be transformed must be explicitly specified too.